Grafo.cpp

```diff
@@ -107,4 +107,170 @@ void Grafo::insertarArista(string origen, string destino, int precio, float tiem
-}
\ No newline at end of file
+
+
+
+Matriz<string>* Grafo::generarRecorrido()
+{
+    Matriz<string>* recorridoMatriz = new Matriz<string>(NONE,elementos,elementos);//MATRIZ NxN CON TODOS NONE
+
+    for(int i = 0; i < elementos; i++)
+    {
+        for(int j = 0 ; j < elementos ; j++ )
+        {
+            if(i != j)
+            {
+                string aeropuerto = vertices->obtenerDato(j);
+                recorridoMatriz->modificarElemento(aeropuerto,i,j);
+            }
+        }
+
+    }
+
+    return recorridoMatriz;
+}
+
+
+Matriz<string>* Grafo::caminoMinimo(Matriz<int>* &precioMatriz)
+{
+    //Inicializo la matriz de Recorrido ...
+    Matriz<string>* recorridoMatriz = generarRecorrido();  ///LIBERAR DESPUES
+
+    //k = es el intermediario -> la letra de base
+    for(int k = 0 ; k < elementos ,k ++)
+    {
+        for(int i = 0 ;  i < elementos ; i ++)
+        {
+            for(int j = 0 ; j < elementos ; j++)
+            {
+                //Distancia acumulada
+                int distancia = precioMatriz->obtenerValor(i,k) + precioMatriz->obtenerValor(k,j);
+
+                if(distancia < precioMatriz->obtenerValor(i,j))
+                    precioMatriz->modificarElemento(distancia,i,j); //Modifico la el valor
+
+                    aeropuerto = vertices->obtenerDato(k);
+                    recorridoMatriz->modificarElemento(aeropuerto,i,j) //Modifico el aeropuerto
+
+            }
+        }
+    }
+    return recorridoMatriz;
+
+}
+
+
+Matriz<string>* Grafo::caminoMinimo(Matriz<float>* &tiempoMatriz)
+{
+    //Inicializo la matriz de Recorrido ...
+    Matriz<string>* recorridoMatriz = generarRecorrido();  ///LIBERAR DESPUES
+
+    //k = es el intermediario -> la letra de base
+    for(int k = 0 ; k < elementos ,k ++)
+    {
+        for(int i = 0 ;  i < elementos ; i ++)
+        {
+            for(int j = 0 ; j < elementos ; j++)
+            {
+                //Distancia acumulada
+                int distancia = tiempoMatriz->obtenerValor(i,k) + tiempoMatriz->obtenerValor(k,j);
+
+                if(distancia < tiempoMatriz->obtenerValor(i,j))
+                    tiempoMatriz->modificarElemento(distancia,i,j); //Modifico la el valor
+
+                    aeropuerto = vertices->obtenerDato(k);
+                    recorridoMatriz->modificarElemento(aeropuerto,i,j) //Modifico el aeropuerto
+
+            }
+        }
+    }
+
+    return recorridoMatriz;
+}
+
+Lista<string>* Grafo::armarEscalas(int posicionOrigen , int posDestino , Matriz<string>* &recorridoMatriz)
+{
+    Lista<string>* escalas = new Lista<string> ; //Lista que posee las escalas
+
+    string escala = recorridoMatriz->obtenerValor(posicionOrigen,posicionDestino);
+
+    while(escala != NONE)
+    {
+        escalas->insertar(escala);
+        int posEscala = vertices->obtenerPosicion(escala);
+        escala =  recorridoMatriz->obtenerValor(posEscala , posicionDestino);
+    }
+
+    return escala;
```

Comment:
Creo que te comiste una `s`, escala es el string, escalas la lista
```suggestion
    return escalas;
```